app/lib/frontend/handlers.dart

```diff
@@ -250,11 +250,7 @@ Future<shelf.Response> searchHandler(shelf.Request request) async {
 
   final Match pkgMatch = _packageRegexp.firstMatch(queryText);
   if (pkgMatch != null) {
-    // only if query attribute was not specified
-    if (packagePrefix == null) {
-      packagePrefix = pkgMatch.group(1).trim();
-      if (packagePrefix.isEmpty) packagePrefix = null;
-    }
+    packagePrefix = pkgMatch.group(1).trim();
```

Comment:
The trim here is not strictly necessary, since it's guaranteed to have only `[_a-z0-9]` characters (and is also not empty, since it's guaranteed to have at least one).